energy-integration/fees-collector/tests/fees_collector_rust_test.rs

```diff
@@ -182,6 +182,113 @@ fn claim_first_week_test() {
         .assert_ok();
 }
 
+#[test]
+fn claim_after_dex_inactive_test() {
+    let rust_zero = rust_biguint!(0);
+    let mut fc_setup = FeesCollectorSetup::new(
+        fees_collector::contract_obj,
+        energy_factory_mock::contract_obj,
+    );
+
+    let first_user = fc_setup.b_mock.create_user_account(&rust_zero);
+    let second_user = fc_setup.b_mock.create_user_account(&rust_zero);
+
+    fc_setup.set_energy(&first_user, 50, 3_000);
+    fc_setup.set_energy(&second_user, 50, 9_000);
+
+    fc_setup.deposit(FIRST_TOKEN_ID, USER_BALANCE).assert_ok();
+    fc_setup
+        .deposit(SECOND_TOKEN_ID, USER_BALANCE / 2)
+        .assert_ok();
+
+    // user claim first week - users only get registered for week 2, without receiving rewards
+    fc_setup.claim(&first_user).assert_ok();
+    fc_setup.claim(&second_user).assert_ok();
+
+    // advance week
+    fc_setup.advance_week();
+    fc_setup.advance_week();
+    fc_setup.advance_week();
+
+    // deposit rewards week 2
+    fc_setup.deposit(FIRST_TOKEN_ID, USER_BALANCE).assert_ok();
+    fc_setup
+        .deposit(SECOND_TOKEN_ID, USER_BALANCE / 2)
+        .assert_ok();
+
+    // decrease user energy
+    fc_setup.set_energy(&first_user, 50, 2_500);
+
+    // first user claims in second week
+    fc_setup.claim(&first_user).assert_ok();
```

Comment:
Add checks for each of the new "week" mappers before and after this claim.

Then, also claim with second user, to see if that works.